expo/expo
# [ios][expo-av] Do not update session configuration when app is switched to background

# Why

When app is switched to background, `expo-av` may set the audio session category to `AVAudioSessionCategorySoloAmbient` in the following call
https://github.com/expo/expo/blob/86589e9cabb9542cb535a4aa28182437640c79f8/packages/expo-av/ios/EXAV/EXAudioSessionManager.m#L159-L174
If all foreground modules are removed, the `_getRequestedCategory` call in `_updateSessionConfiguration` will always return `AVAudioSessionCategorySoloAmbient`. This means application will interrupt other apps audio sessions as `expo-av` is not deactivating its audio session anymore.

# How

Simply remove the `_updateSessionConfiguration` call in `moduleDidBackground`. It does not really make sense there. When app is switched to background, moduleDidBackground is supposed to be called by all foreground modules, one by one. And eventually you get an empty foreground module list. So it will always update the audio session category and stops non-mixable audio playback in other apps such as Spotify.

When app is put back to foreground, the correct audio session category will be handled automatically by `_updateSessionConfiguration` call in `moduleDidForeground`. This fixes #19042.

# Test Plan

Manual test with `bare-expo` and Spotify on real device (it includes app switching so I am not sure how to automate it).
- Build & install `bare-expo` on real iOS device.
- Open Spotify and play some music.
- Switch to `bare-expo` and select Audio in API tests.
- Ensure "Mix with others" and selected and start playback in test.
- Put `bare-expo` into background.
- Spotify shouldn't stop playback (without the changes it stops playback).

# Checklist
